src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java

```diff
@@ -464,15 +464,23 @@ private void check_remote_url(final String repositoryName) throws InterruptedExc
         assertTrue("remote URL has not been updated", remotes.contains(localMirror()));
     }
 
+    private Collection<String> getBranchNames(Set<Branch> branches) {
+        return Collections2.transform(branches, GitObject::getName);
+    }
+
+    private void assertBranchCount(Set<Branch> branches, int expectedCount) {
+        assertEquals("Unexpected branch count in " + getBranchNames(branches), expectedCount, branches.size());
+    }
+
     private void assertBranchesExist(Set<Branch> branches, String ... names) throws InterruptedException {
-        Collection<String> branchNames = Collections2.transform(branches, GitObject::getName);
+        Collection<String> branchNames = getBranchNames(branches);
         for (String name : names) {
             assertTrue(name + " branch not found in " + branchNames, branchNames.contains(name));
         }
     }
 
     private void assertBranchesNotExist(Set<Branch> branches, String ... names) throws InterruptedException {
```

Comment:
This is similar to [`assertBranchesExist(...)`](https://github.com/jenkinsci/git-client-plugin/pull/351/files#diff-963c705aa3f7bb6ac484408977597efbR475) with Hamcrest's [not(...)](http://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#not(org.hamcrest.Matcher)) sneaked in.